copystructure.go

```diff
@@ -73,6 +86,11 @@ type Config struct {
 	// Copiers is a map of types associated with a CopierFunc. Use the global
 	// Copiers map if this is nil.
 	Copiers map[reflect.Type]CopierFunc
+
+	// ShallowCopiers is a map of pointer types that when they are
+	// shallow copied no matter where they are encountered. Use the
+	// global ShallowCopiers if this is nil.
+	ShallowCopiers map[reflect.Type]struct{}
```

Comment:
If the map is used only for checking existence, is it worth making the value type `bool` instead of `struct{}`?

Reply:
I generally prefer struct here to communicate intent more than anything (the bool here would be meaningless), but if this is something that we'd prefer to have changed to bool here I'm fine with that.

Reply:
If its only existence checking, I prefer `struct{}` as well. That is more idiomatic for Go structs. It has a slight performance benefit as well but that doesn't matter here at all (struct takes zero bytes, bool takes one).

Reply:
I see, all very good points! 👍🏻